tests/e2e/no_hci_mesh_rwx_singlevc_topology_disruptive.go

```diff
@@ -0,0 +1,726 @@
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		ginkgo.By("Set allowed topology for pod node selector terms")
+		if topologySetupType == "Level2" {
+			/* Setting zone1 for pod node selector terms */
+			keyValues := []KeyValue{
+				{KeyIndex: 0, ValueIndex: 0}, // Key: k8s-region, Value: region1
+				{KeyIndex: 1, ValueIndex: 0}, // Key: k8s-zone, Value: zone1
+			}
+			allowedTopologyForPod, err = getAllowedTopologyForLevel2(allowedTopologies, keyValues)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			nodeSelectorTerms, err = getNodeSelectorMapForDeploymentPods(allowedTopologyForPod)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		} else if topologySetupType == "Level5" {
+			/* setting rack1(cluster1) as node selector terms for deployment pod creation */
+			allowedTopologyForPod = getTopologySelector(topologyAffinityDetails, topologyCategories,
+				topologyLength, leafNode, leafNodeTag0)[4:]
+			nodeSelectorTerms, err = getNodeSelectorMapForDeploymentPods(allowedTopologyForPod)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}
+
+		ginkgo.By("Create Deployments and specify node selector terms only for deployment1, " +
+			"rest all other deployment pod should get spread across all AZ")
+		for i := 0; i < len(pvclaims); i++ {
+			if i == 0 {
+				deploymentList, podsNodeSelectorterm, err = createVerifyAndScaleDeploymentPods(ctx, client, namespace,
+					replica, false, labelsMap, pvclaims[i], nodeSelectorTerms, execRWXCommandPod,
+					nginxImage, false, nil, createDepItr)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+				depl = append(depl, deploymentList...)
+			} else {
+				deploymentList, _, err := createVerifyAndScaleDeploymentPods(ctx, client, namespace, replica, false, labelsMap,
+					pvclaims[i], nil, execRWXCommandPod, nginxImage, false, nil, createDepItr)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+				depl = append(depl, deploymentList...)
+			}
+		}
+		defer func() {
+			for i := 0; i < len(depl); i++ {
+				framework.Logf("Delete deployment set")
+				err = client.AppsV1().Deployments(namespace).Delete(ctx, depl[i].Name, *metav1.NewDeleteOptions(0))
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			}
+		}()
+
+		ginkgo.By("Verify deployment pods are scheduled on a specified node selector terms")
+		err = verifyDeploymentPodNodeAffinity(ctx, client, namespace, allowedTopologyForPod, podsNodeSelectorterm)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		ginkgo.By("Bring down all esxi hosts in Az2")
+		hostsInCluster := getHostsByClusterName(ctx, clusterComputeResource, topologyClusterList[1])
+		powerOffHostsList = powerOffEsxiHostByCluster(ctx, &e2eVSphere, topologyClusterList[1],
+			len(hostsInCluster))
+		defer func() {
+			ginkgo.By("Bring up all ESXi host which were powered off in zone2")
+			for i := 0; i < len(powerOffHostsList); i++ {
+				powerOnEsxiHostByCluster(powerOffHostsList[i])
+			}
+			k8sMasterIPs := getK8sMasterIPs(ctx, client)
```

Comment:
write a util to use wherever required to check k8s nodes

Reply:
Ack